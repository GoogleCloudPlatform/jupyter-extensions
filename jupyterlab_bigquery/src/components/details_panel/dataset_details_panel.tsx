import * as React from 'react';

import {
  DatasetDetailsService,
  DatasetDetails,
} from './service/list_dataset_details';
import { Header } from '../shared/header';
import LoadingPanel from '../loading_panel';
import { DetailsPanel } from './details_panel';
import { stylesheet } from 'typestyle';
import { formatDate, formatMs } from '../../utils/formatters';
import { BASE_FONT } from 'gcp_jupyterlab_shared';

export const localStyles = stylesheet({
  body: {
    marginRight: '24px',
    marginLeft: '24px',
    flex: 1,
    minHeight: 0,
    overflowY: 'auto',
    overflowX: 'hidden',
  },
  container: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    ...BASE_FONT,
  },
});

interface Props {
  datasetDetailsService: DatasetDetailsService;
  isVisible: boolean;
  dataset_id: string;
}

interface State {
  hasLoaded: boolean;
  isLoading: boolean;
  details: DatasetDetails;
  rows: DetailRow[];
}

interface DetailRow {
  name: string;
  value: string;
}

export default class DatasetDetailsPanel extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      hasLoaded: false,
      isLoading: false,
      details: { details: {} } as DatasetDetails,
      rows: [],
    };
  }

  componentDidUpdate(prevProps: Props) {
    const isFirstLoad =
      !(this.state.hasLoaded || prevProps.isVisible) && this.props.isVisible;
    if (isFirstLoad) {
      this.getDetails();
    }
  }

  private async getDetails() {
    try {
      this.setState({ isLoading: true });
      const details = await this.props.datasetDetailsService.listDatasetDetails(
        this.props.dataset_id
      );

      const detailsObj = details.details;
      const rows = [
        { name: 'Dataset ID', value: detailsObj.id },
        { name: 'Created', value: formatDate(detailsObj.date_created) },
        {
          name: 'Default table expiration',
          value: detailsObj.default_expiration
            ? formatMs(detailsObj.default_expiration)
            : 'Never',
        },
        {
          name: 'Last modified',
          value: formatDate(detailsObj.last_modified),
        },
        {
          name: 'Data location',
          value: detailsObj.location ? detailsObj.location : 'None',
        },
      ];

      this.setState({ hasLoaded: true, details, rows });
    } catch (err) {
      console.warn('Error retrieving dataset details', err);
    } finally {
      this.setState({ isLoading: false });
    }
  }

  render() {
    if (this.state.isLoading) {
      return <LoadingPanel />;
    } else {
      return (
        <div className={localStyles.container}>
          <Header>{this.props.dataset_id}</Header>
          <div className={localStyles.body}>
            <DetailsPanel
              details={this.state.details.details}
              rows={this.state.rows}
              detailsType="DATASET"
            />
          </div>
        </div>
      );
    }
  }
}
