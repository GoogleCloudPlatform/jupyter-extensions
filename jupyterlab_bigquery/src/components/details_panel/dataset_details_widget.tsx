import { ReactWidget } from '@jupyterlab/apputils';
import * as React from 'react';

import { DatasetDetailsService } from './service/list_dataset_details';
import DatasetDetailsPanel from './dataset_details_panel';

/** Widget to be registered in the main panel. */
export class DatasetDetailsWidget extends ReactWidget {
  id = 'dataset-details-widget';

  constructor(
    private readonly service: DatasetDetailsService,
    private readonly dataset_id: string,
    private readonly name: string
  ) {
    super();
    this.title.iconClass = 'jp-Icon jp-Icon-20 jp-DatasetIcon';
    this.title.caption = `Dataset Details for ${this.dataset_id}`;
    this.title.label = this.name;
    this.title.closable = true;
  }

  render() {
    return (
      <DatasetDetailsPanel
        isVisible={this.isVisible}
        dataset_id={this.dataset_id}
        datasetDetailsService={this.service}
      />
    );
  }
}
