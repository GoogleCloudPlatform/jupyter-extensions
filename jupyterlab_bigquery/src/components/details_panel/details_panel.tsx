import * as React from 'react';

import {
  Grid,
  Chip,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  withStyles,
} from '@material-ui/core';
import { stylesheet } from 'typestyle';

import ReadOnlyEditor from '../shared/read_only_editor';
import { SchemaField } from './service/list_table_details';
import { ModelSchema } from './service/list_model_details';
import { StripedRows } from '../shared/striped_rows';

export const localStyles = stylesheet({
  title: {
    fontSize: '16px',
    marginBottom: '10px',
  },
  panel: {
    backgroundColor: 'white',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  detailsBody: {
    fontSize: '13px',
    marginTop: '24px',
    marginBottom: '24px',
  },
  labelContainer: {
    display: 'flex',
    flexWrap: 'wrap',
    '& > *': {
      marginRight: '8px',
      marginBottom: '8px',
    },
  },
  rowTitle: {
    width: '200px',
  },
  row: {
    display: 'flex',
    padding: '6px',
  },
});

export const TableHeadCell: React.ComponentType<any> = withStyles({
  root: {
    backgroundColor: '#f0f0f0',
  },
})(TableCell);

const StyledChip = withStyles({
  root: {
    color: '#1967D2',
    backgroundColor: 'rgba(25, 103, 210, 0.1)',
  },
})(Chip);

const formatFieldName = name => {
  if (name.includes('.')) {
    const child = name.substr(name.lastIndexOf('.') + 1);
    const parents = name.substr(0, name.lastIndexOf('.') + 1);
    return (
      <div>
        {parents} <b>{child}</b>
      </div>
    );
  } else {
    return <b>{name}</b>;
  }
};

interface SharedDetails {
  id: string;
  description: string;
  labels: string[];
  name: string;
  schema?: SchemaField[];
  query?: string;
  schema_labels?: ModelSchema[];
  feature_columns?: ModelSchema[];
}

interface Props {
  details: SharedDetails;
  rows: any[];
  // TODO(cxjia): figure out a shared typing for these rows
  detailsType: 'DATASET' | 'TABLE' | 'VIEW' | 'MODEL';
}

const getTitle = type => {
  switch (type) {
    case 'DATASET':
      return 'Dataset info';
    case 'TABLE':
      return 'Table info';
    case 'VIEW':
      return 'View info';
    case 'MODEL':
      return 'Model details';
  }
};

export const DetailsPanel: React.SFC<Props> = props => {
  const { details, rows, detailsType } = props;

  return (
    <div className={localStyles.panel}>
      <div className={localStyles.detailsBody}>
        <Grid container spacing={4}>
          <Grid item xs={6}>
            <div className={localStyles.title}>Description</div>
            <div>{details.description ? details.description : 'None'}</div>
          </Grid>

          <Grid item xs={6}>
            <div>
              <div className={localStyles.title}>Labels</div>
              {details.labels ? (
                <div className={localStyles.labelContainer}>
                  {details.labels.map((value, index) => {
                    return (
                      <StyledChip size="small" key={index} label={value} />
                    );
                  })}
                </div>
              ) : (
                'None'
              )}
            </div>
          </Grid>

          <Grid item xs={12}>
            <div className={localStyles.title}>{getTitle(detailsType)}</div>
            <StripedRows rows={rows} />
          </Grid>
        </Grid>

        {(detailsType === 'TABLE' || detailsType === 'VIEW') && (
          <div>
            <div className={localStyles.title} style={{ marginTop: '32px' }}>
              Schema
            </div>
            {details.schema && details.schema.length > 0 ? (
              <Table
                size="small"
                style={{ width: 'auto', tableLayout: 'auto' }}
              >
                <TableHead>
                  <TableRow>
                    <TableHeadCell>Field name</TableHeadCell>
                    <TableHeadCell>Type</TableHeadCell>
                    <TableHeadCell>Mode</TableHeadCell>
                    <TableHeadCell>Description</TableHeadCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {details.schema.map((field, index) => {
                    return (
                      <TableRow key={`schema_row_${index}`}>
                        <TableCell>{formatFieldName(field.name)}</TableCell>
                        <TableCell>{field.type}</TableCell>
                        <TableCell>{field.mode}</TableCell>
                        <TableCell>{field.description ?? ''}</TableCell>
                      </TableRow>
                    );
                  })}
                </TableBody>
              </Table>
            ) : (
              'Table does not have a schema.'
            )}
          </div>
        )}

        {detailsType === 'VIEW' && (
          <div>
            <div className={localStyles.title} style={{ marginTop: '32px' }}>
              Query
            </div>
            <ReadOnlyEditor query={details.query} />
          </div>
        )}

        {detailsType === 'MODEL' && (
          <div>
            <div className={localStyles.title} style={{ marginTop: '32px' }}>
              Label columns
            </div>
            <div>
              {details.schema_labels && details.schema_labels.length > 0 ? (
                <Table
                  size="small"
                  style={{ width: 'auto', tableLayout: 'auto' }}
                >
                  <TableHead>
                    <TableRow>
                      <TableHeadCell>Field name</TableHeadCell>
                      <TableHeadCell>Type</TableHeadCell>
                    </TableRow>
                  </TableHead>
                  <TableBody>
                    {details.schema_labels.map((field, index) => {
                      return (
                        <TableRow key={`schema_label_row_${index}`}>
                          <TableCell>{field.name}</TableCell>
                          <TableCell>{field.type}</TableCell>
                        </TableRow>
                      );
                    })}
                  </TableBody>
                </Table>
              ) : (
                'Model does not have any label columns.'
              )}
            </div>

            <div className={localStyles.title} style={{ marginTop: '24px' }}>
              Feature columns
            </div>
            <div>
              {details.feature_columns && details.feature_columns.length > 0 ? (
                <Table
                  size="small"
                  style={{ width: 'auto', tableLayout: 'auto' }}
                >
                  <TableHead>
                    <TableRow>
                      <TableHeadCell>Field name</TableHeadCell>
                      <TableHeadCell>Type</TableHeadCell>
                    </TableRow>
                  </TableHead>
                  <TableBody>
                    {details.feature_columns.map((field, index) => {
                      return (
                        <TableRow key={`schema_feature_row_${index}`}>
                          <TableCell>{field.name}</TableCell>
                          <TableCell>{field.type}</TableCell>
                        </TableRow>
                      );
                    })}
                  </TableBody>
                </Table>
              ) : (
                'Model does not have any feature columns.'
              )}
            </div>
          </div>
        )}
      </div>
    </div>
  );
};
