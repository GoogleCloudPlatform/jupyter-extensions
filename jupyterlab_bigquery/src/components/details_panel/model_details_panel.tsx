/* eslint-disable @typescript-eslint/camelcase */
import * as React from 'react';
import { Button, Select, MenuItem, withStyles } from '@material-ui/core';
import { Code } from '@material-ui/icons';

import {
  ModelDetailsService,
  ModelDetails,
} from './service/list_model_details';
import LoadingPanel from '../loading_panel';
import { DetailsPanel } from './details_panel';
import { Header } from '../shared/header';
import { QueryEditorTabWidget } from '../query_editor/query_editor_tab/query_editor_tab_widget';
import { WidgetManager } from '../../utils/widgetManager/widget_manager';
import { generateQueryId } from '../../reducers/queryEditorTabSlice';
import { localStyles } from './dataset_details_panel';
import { formatDate } from '../../utils/formatters';
import { getStarterQuery } from '../../utils/starter_queries';
import { gColor } from '../shared/styles';

interface Props {
  modelDetailsService: ModelDetailsService;
  isVisible: boolean;
  modelId: string;
  modelName: string;
}

interface State {
  hasLoaded: boolean;
  isLoading: boolean;
  panelIsLoading: boolean;
  details: ModelDetails;
  rows: DetailRow[];
  trainingRows: any[];
  currentRun: number;
  loadedRuns: any;
}

interface DetailRow {
  name: string;
  value: string | number;
}

const displayOptionNames = {
  actual_iterations: 'Actual iterations',
  data_split_column: 'Data split column',
  data_split_eval_fraction: 'Data split eval fraction',
  data_split_method: 'Data split method',
  distance_type: 'Distance type',
  early_stop: 'Early stop',
  initial_learn_rate: 'Initial learn rate',
  input_label_columns: 'Input label columns',
  kmeans_initialization_column: 'Kmeans initialization column',
  kmeans_initialization_method: 'Centroids initialization method',
  l1_regularization: 'L1 regularization',
  l2_regularization: 'L2 regularization',
  label_class_weights: 'Label class weights',
  learn_rate: 'Learn rate',
  learn_rate_strategy: 'Learn rate strategy',
  loss_type: 'Loss type',
  max_iterations: 'Max allowed iterations',
  min_relative_progress: 'Min relative progress',
  model_uri: 'Model uri',
  num_clusters: 'Number of clusters',
  optimization_strategy: 'Optimization strategy',
  warm_start: 'Warm start',
};

const StyledMenuItem = withStyles({
  root: {
    color: 'var(--jp-ui-font-color1)',
    backgroundColor: 'var(--jp-layout-color0)',
    '&$selected': {
      backgroundColor: 'var(--jp-layout-color2)',
      '&:hover': {
        backgroundColor: 'var(--jp-layout-color2)',
      },
    },
    '&:hover': {
      backgroundColor: 'var(--jp-layout-color2)',
    },
  },
  selected: {},
})(MenuItem);

// TODO: style for dark mode. Currently the container is still constant white.
export const StyledSelect = withStyles({
  root: {
    marginLeft: '36px',
    color: 'var(--jp-ui-font-color1)',
  },
  icon: {
    color: 'var(--jp-ui-font-color1)',
  },
})(Select);

export default class ModelDetailsPanel extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      hasLoaded: false,
      isLoading: false,
      panelIsLoading: false,
      details: { details: {} } as ModelDetails,
      rows: [],
      currentRun: null,
      trainingRows: [],
      loadedRuns: {},
    };
  }

  async componentDidUpdate(prevProps: Props) {
    const isFirstLoad =
      !(this.state.hasLoaded || prevProps.isVisible) && this.props.isVisible;
    if (isFirstLoad) {
      await this.getDetails();
      this.getTrainingRunDetails(
        this.state.details.details.training_runs.length - 1
      );
    }
  }

  private async getDetails() {
    try {
      this.setState({ isLoading: true });
      const details = await this.props.modelDetailsService.listModelDetails(
        this.props.modelId
      );

      const detailsObj = details.details;
      const rows = [
        { name: 'Model ID', value: detailsObj.id },
        { name: 'Date created', value: formatDate(detailsObj.date_created) },
        {
          name: 'Model expiration',
          value: detailsObj.expires ? formatDate(detailsObj.expires) : 'Never',
        },
        {
          name: 'Date modified',
          value: formatDate(detailsObj.last_modified),
        },
        {
          name: 'Data location',
          value: detailsObj.location ? detailsObj.location : 'None',
        },
        {
          name: 'Model type',
          value: detailsObj.model_type,
        },
      ];
      this.setState({
        hasLoaded: true,
        details,
        rows,
        currentRun: detailsObj.training_runs.length - 1,
      });
    } catch (err) {
      console.warn('Error retrieving model details', err);
    } finally {
      this.setState({ isLoading: false });
    }
  }

  private async getTrainingRunDetails(runIndex: number) {
    try {
      this.setState({ panelIsLoading: true });

      if (!this.state.loadedRuns[runIndex]) {
        const details = await this.props.modelDetailsService.getTrainingRunDetails(
          this.props.modelId,
          runIndex
        );

        const trainingRows = Object.entries(details.details).map(pair => {
          return { name: displayOptionNames[pair[0]], value: pair[1] };
        });

        const updatedRuns = { ...this.state.loadedRuns };
        updatedRuns[runIndex] = trainingRows;
        this.setState({ loadedRuns: updatedRuns });
      }
    } catch (err) {
      console.warn('Error retrieving model training run details', err);
    } finally {
      this.setState({ panelIsLoading: false });
    }
  }

  render() {
    if (this.state.isLoading) {
      return <LoadingPanel />;
    } else {
      return (
        <div className={localStyles.container}>
          <Header>
            <div>
              {this.props.modelName}
              {this.state.details.details.training_runs &&
              this.state.details.details.training_runs.length > 1 ? (
                <StyledSelect
                  value={this.state.currentRun}
                  onChange={event => {
                    this.setState({ currentRun: event.target.value as number });
                    this.getTrainingRunDetails(event.target.value as number);
                  }}
                  disableUnderline
                >
                  {this.state.details.details.training_runs &&
                    this.state.details.details.training_runs.map(
                      (date, index) => {
                        return (
                          <StyledMenuItem
                            value={index}
                            key={`training_run_${index}`}
                          >
                            {index + 1} ({formatDate(date)})
                          </StyledMenuItem>
                        );
                      }
                    )}
                </StyledSelect>
              ) : (
                undefined
              )}
            </div>
            <Button
              onClick={() => {
                const queryId = generateQueryId();
                WidgetManager.getInstance().launchWidget(
                  QueryEditorTabWidget,
                  'main',
                  queryId,
                  undefined,
                  [queryId, getStarterQuery('MODEL', this.props.modelId)]
                );
              }}
              startIcon={<Code />}
              style={{
                textTransform: 'none',
                color: gColor('BLUE'),
              }}
            >
              Query model
            </Button>
          </Header>

          <div className={localStyles.body}>
            {this.state.panelIsLoading ? (
              <LoadingPanel />
            ) : (
              <DetailsPanel
                details={this.state.details.details}
                rows={this.state.rows}
                trainingRows={this.state.loadedRuns[this.state.currentRun]}
                detailsType="MODEL"
              />
            )}
          </div>
        </div>
      );
    }
  }
}
