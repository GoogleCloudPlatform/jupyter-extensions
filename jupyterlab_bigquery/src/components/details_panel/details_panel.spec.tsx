import * as React from 'react';
import { shallow } from 'enzyme';
import { Settings } from 'luxon';

import { DetailsPanel } from './details_panel';
import { SchemaTable, ModelSchemaTable } from '../shared/schema_table';
import {
  FakeDatasetDetailsFull,
  FakeDatasetDetailsEmpty,
  FakeTableDetailsFull,
  FakeTableDetailsEmpty,
  FakeViewDetailsFull,
  FakeModelDetailsFullSchema,
  FakeModelDetailsEmptySchema,
} from './test_helpers';

describe('DetailsPanel', () => {
  const fakeRows = [
    { name: 'name', value: 'value' },
    { name: 'another name', value: 'another value' },
  ];

  const fakeTrainingRows = [
    { name: 'training option', value: 'training value' },
  ];

  beforeEach(() => {
    Settings.defaultZoneName = 'America/Los_Angeles';
  });

  afterEach(() => {
    Settings.defaultZoneName = 'local';
  });

  it('Renders for datasets with description and labels', () => {
    const component = shallow(
      <DetailsPanel
        details={FakeDatasetDetailsFull}
        rows={fakeRows}
        detailsType="DATASET"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('Renders for datasets without description and labels', () => {
    const component = shallow(
      <DetailsPanel
        details={FakeDatasetDetailsEmpty}
        rows={fakeRows}
        detailsType="DATASET"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('Renders for tables with all values populated', () => {
    const component = shallow(
      <DetailsPanel
        details={FakeTableDetailsFull}
        rows={fakeRows}
        detailsType="TABLE"
      />
    );
    expect(component).toMatchSnapshot();
    expect(component.find(SchemaTable).exists()).toBeTruthy();
  });

  it('Renders for tables without schema', () => {
    const component = shallow(
      <DetailsPanel
        details={FakeTableDetailsEmpty}
        rows={fakeRows}
        detailsType="TABLE"
      />
    );
    expect(component.find(SchemaTable).exists()).toBeFalsy();
    expect(component).toMatchSnapshot();
  });

  it('Renders a read-only editor for views', () => {
    const component = shallow(
      <DetailsPanel
        details={FakeViewDetailsFull}
        rows={fakeRows}
        detailsType="VIEW"
      />
    );
    expect(component.find('ReadOnlyEditor').exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('Renders for models with label columns and feature columns', () => {
    const component = shallow(
      <DetailsPanel
        details={FakeModelDetailsFullSchema}
        rows={fakeRows}
        trainingRows={fakeTrainingRows}
        detailsType="MODEL"
      />
    );
    expect(component).toMatchSnapshot();
    expect(component.find(ModelSchemaTable).exists()).toBeTruthy();
  });

  it('Renders for models without label columns or feature columns', () => {
    const component = shallow(
      <DetailsPanel
        details={FakeModelDetailsEmptySchema}
        rows={fakeRows}
        trainingRows={fakeTrainingRows}
        detailsType="MODEL"
      />
    );
    expect(component.find(ModelSchemaTable).exists()).toBeFalsy();
    expect(component).toMatchSnapshot();
  });
});
