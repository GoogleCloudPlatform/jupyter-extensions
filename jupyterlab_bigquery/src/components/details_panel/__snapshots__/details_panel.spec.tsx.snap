// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPanel Renders a read-only editor for views 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          description of view
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          <div
            className="labelContainer_f14ovan"
          >
            <WithStyles(Component)
              darkmode={false}
              key="0"
              label="label: value"
              size="small"
            />
            <WithStyles(Component)
              darkmode={false}
              key="1"
              label="another label: another value"
              size="small"
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          View info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "name",
                "value": "value",
              },
              Object {
                "name": "another name",
                "value": "another value",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Schema
        </div>
        Table does not have a schema.
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Query
        </div>
        <ReadOnlyEditor
          query="SELECT * FROM \`projectId.datasetId.tableId\` LIMIT 100"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for datasets with description and labels 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          description of dataset
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          <div
            className="labelContainer_f14ovan"
          >
            <WithStyles(Component)
              darkmode={false}
              key="0"
              label="label: value"
              size="small"
            />
            <WithStyles(Component)
              darkmode={false}
              key="1"
              label="another label: another value"
              size="small"
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Dataset info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "name",
                "value": "value",
              },
              Object {
                "name": "another name",
                "value": "another value",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for datasets without description and labels 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Dataset info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "name",
                "value": "value",
              },
              Object {
                "name": "another name",
                "value": "another value",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for models with label columns and feature columns 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          description of model
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          <div
            className="labelContainer_f14ovan"
          >
            <WithStyles(Component)
              darkmode={false}
              key="0"
              label="label: value"
              size="small"
            />
            <WithStyles(Component)
              darkmode={false}
              key="1"
              label="another label: another value"
              size="small"
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Model details
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "name",
                "value": "value",
              },
              Object {
                "name": "another name",
                "value": "another value",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Training options
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "training option",
                "value": "training value",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Label columns
        </div>
        <div>
          <Component
            schema={
              Array [
                Object {
                  "name": "predicted_field",
                  "type": "FLOAT",
                },
              ]
            }
          />
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Feature columns
        </div>
        <div>
          <Component
            schema={
              Array [
                Object {
                  "name": "feature_0",
                  "type": "INTEGER",
                },
                Object {
                  "name": "feature_1",
                  "type": "FLOAT",
                },
              ]
            }
          />
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for models without label columns or feature columns 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Model details
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "name",
                "value": "value",
              },
              Object {
                "name": "another name",
                "value": "another value",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Training options
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "training option",
                "value": "training value",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Label columns
        </div>
        <div>
          Model does not have any label columns.
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Feature columns
        </div>
        <div>
          Model does not have any feature columns.
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for tables with all values populated 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          description of table
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          <div
            className="labelContainer_f14ovan"
          >
            <WithStyles(Component)
              darkmode={false}
              key="0"
              label="label: value"
              size="small"
            />
            <WithStyles(Component)
              darkmode={false}
              key="1"
              label="another label: another value"
              size="small"
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Table info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "name",
                "value": "value",
              },
              Object {
                "name": "another name",
                "value": "another value",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Schema
        </div>
        <Component
          schema={
            Array [
              Object {
                "description": "this field is a record",
                "mode": "NULLABLE",
                "name": "record",
                "type": "RECORD",
              },
              Object {
                "description": "this field is a string and a child",
                "mode": "NULLABLE",
                "name": "record.child",
                "type": "STRING",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for tables without schema 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Table info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "name",
                "value": "value",
              },
              Object {
                "name": "another name",
                "value": "another value",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Schema
        </div>
        Table does not have a schema.
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;
